### PR TITLE
NEW FEATURE: Reader service for country and company with unit test

### PR DESCRIPTION
**Pull request by Raziel-777**

### CODE REVIEW

File: java/src/com/mapJava/main.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class main {
+
+    public static void main(String[] args) throws IOException {
+        List<String[]> toto = new ArrayList<>();
+        toto = Reader.readCompany("data/testCompany1579Lines.csv", 1);
```
Comment (Niavliss) [inline java/src/com/mapJava/main.java]: Who is toto ?

File: java/src/com/mapJava/services/Reader.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+public class Reader {
+
+    /**
+     * @param file:      path file to parse
+     * @param lineStart: line to start parsing
+     * @return List of array of string in order to create Company object
+     * @throws IOException: take a look at the IOException doc
+     */
+    public static List<String[]> readCompany(String file, int lineStart) throws IOException {
+        CSVReader reader = null;
```
Comment (Roms07) [inline java/src/com/mapJava/services/Reader.java]: why use a CVSReader ?

File: java/src/com/mapJava/services/Reader.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+    /**
+     * @param file:      path file to parse
+     * @param lineStart: line to start parsing
+     * @return List of array of string in order to create Company object
+     * @throws IOException: RTFM
+     */
+    public static List<String[]> readCountry(String file, int lineStart) throws IOException {
+        CSVReader reader = null;
+        FileReader fileReader;
```
Comment (Roms07) [inline java/src/com/mapJava/services/Reader.java]: why use a CVSReader and a FileReader ?

File: java/src/com/mapJava/services/Reader.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        List<String[]> resultCsv = new ArrayList<>();
+        int i = 0;
+        try {
+            fileReader = new FileReader(file);
+            reader = new CSVReader(fileReader, ',', '"', lineStart);
+            String[] nextLine;
+            String[] sendLine;
+            while ((nextLine = reader.readNext()) != null && i < 1000) {
```
Comment (Roms07) [inline java/src/com/mapJava/services/Reader.java]: why " != null", not another way to test if there's another line ?

Comment (Raziel-777) [reply java/src/com/mapJava/services/Reader.java]: no

File: java/src/com/mapJava/services/Reader.java (hunk 3 above)
Comment (Raziel-777) [reply java/src/com/mapJava/services/Reader.java]: Not rienvent the wheel

File: java/src/com/mapJava/services/Reader.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
+package com.mapJava.services;
+
+import com.opencsv.CSVReader;
```
Comment (mbaechler) [inline java/src/com/mapJava/services/Reader.java]: now that you managed to use CSVReader (that would be a perfect solution in enterprise context), I'd like you write your own parser, please

**Comment by mbaechler** [comment]
@Niavliss  : you are supposed to review that PR for tomorrow morning

**Comment by Niavliss** [comment]
Done yesterday

File: java/test/com/mapJava/services/TestReader.java [hunk 6]
```diff
@@ -0,0 +1,419 @@
...
+        //Assertion for size list
+        assertThat(resultCsv).hasSize(9);
+        //Assertion data
+        for (int j = 0; j < 9; j++) {
+            for (int z = 0; z < 2; z++) {
+                assertThat(resultCsv.get(j)[z]).isEqualTo(expectedResult.get(j)[z]);
+            }
+        }
+    }
+}
```
Comment (Roms07) [inline java/test/com/mapJava/services/TestReader.java]: maybe you should factor your code and use the name of the file to read like a parameter of the methods...

File: java/test/com/mapJava/services/TestReader.java [hunk 7]
```diff
@@ -0,0 +1,419 @@
...
+
+        String[] three = new String[3];
+        three[0] = "3";
+        three[1] = "Allemagne";
+        three[2] = "DE";
+
+        expectedResult.add(one);
+        expectedResult.add(two);
+        expectedResult.add(three);
+
```
[the remark this answers is not in the document]
Comment (Raziel-777) [reply java/test/com/mapJava/services/TestReader.java]: I prefer to separate reader and factory. I will rename file by readCsvCountry

File: java/test/com/mapJava/services/TestReader.java (hunk 6 above)
Comment (Raziel-777) [reply java/test/com/mapJava/services/TestReader.java]: Yes i can do this

File: java/test/com/mapJava/services/TestReader.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (Raziel-777) [reply java/test/com/mapJava/services/TestReader.java]: No i will rename by just read, one class for all readers

File: java/test/com/mapJava/services/TestReader.java [hunk 8]
```diff
@@ -0,0 +1,419 @@
...
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.*;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+class TestReader {
```
Comment (mbaechler) [inline java/test/com/mapJava/services/TestReader.java]: should be ReaderTest, (class under test + Test suffix)

**Pull request closed without merging**